### ISSUE TITLE
Error: addressBook is undefined

### ISSUE THREAD

**Issue by kyrofa** [nextcloud/contacts#300]
After installing Nextcloud v12(.0.1, .0.2, or daily):

### Steps to reproduce
1. Install Nextcloud fresh (using the snap, personally)
2. Install Contacts app
3. Hit "New contact"

### Expected behaviour
A new contact is created

### Actual behaviour
Nothing happens, with no indication of an error. Opening Firefox's console you see this:

![image](https://user-images.githubusercontent.com/946674/29724753-92a8eb74-897e-11e7-8df6-d5c224e81ef7.png)

### Server configuration
**Operating system**: Ubuntu 16.04

**Web server:** Apache 2.4

**Database:** MySQL 5.7

**PHP version:** 7.0

**Nextcloud version:** (see Nextcloud admin page) 12 daily stable

**Contacts version:** (see Nextcloud apps page) 1.5.3

**Updated from an older Nextcloud or fresh install:** Fresh

**Signing status:**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.5.2
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.3
  - comments: 1.2.0
  - contacts: 1.5.3
  - dav: 1.3.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - workflowengine: 1.2.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_external
  - user_ldap
```

**Nextcloud configuration:**

```
{
    "system": {
        "apps_paths": [
            {
                "path": "\/snap\/nextcloud\/current\/htdocs\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/snap\/nextcloud\/current\/nextcloud\/extra-apps",
                "url": "\/extra-apps",
                "writable": true
            }
        ],
        "supportedDatabases": [
            "mysql"
        ],
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/tmp\/sockets\/redis.sock",
            "port": 0
        },
        "instanceid": "ocqxd47q8cra",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "pandora.local"
        ],
        "datadirectory": "\/var\/snap\/nextcloud\/common\/nextcloud\/data",
        "overwrite.cli.url": "http:\/\/pandora.local",
        "dbtype": "mysql",
        "version": "12.0.1.5",
        "dbname": "nextcloud",
        "dbhost": "localhost:\/tmp\/sockets\/mysql.sock",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true
    }
}
```

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox v55

**Operating system:** Ubuntu 16.04

**CardDAV-clients:** None

**Comment by skjnldsv** [comment]
Do you have an addressbook in the settings list?

**Comment by kyrofa** [comment]
Sorry, I neglected to specify: no. I need to go down the wheel icon and create one first, and then this works as expected. However, this step was not necessary on Nextcloud v11. Is it a regression?

**Comment by skjnldsv** [comment]
It is a bug from the migration and was fix in the (incoming) next version #91
